GetActiveDHCPLeasesWithAssociations crashes docker app

unpoller  2.34.0 and I use unifi ux7 with last official network app. I installed the unpoller docker on 2026-02-18 and I see several errors since I installed it but the docker never crashes since today:

errors frequency:

<img width="1868" height="813" alt="Image" src="https://github.com/user-attachments/assets/64a198cd-c738-44fa-8901-bc9125b1c783" />

```yaml
[unpoller] 2026/03/04 18:53:38 [ERROR] Device tags endpoint not available (e.g. 404). To see which endpoints your controller supports, run endpoint discovery (e.g. unpoller --discover). 
```

```yaml
[unpoller] 2026/03/04 20:50:37 [ERROR] failed to fetch clients for DHCP lease association: unexpected end of JSON input 
```

```yaml
[unpoller] 2026/03/04 00:05:52 [ERROR] failed to fetch devices for DHCP lease association: https://192.168.0.1/proxy/network/api/s/default/stat/device: 401 Unauthorized: invalid status code from server 
```

```yaml
[unpoller] 2026/03/07 14:46:07 [ERROR] Failed to collect metrics from controller https://192.168.0.1/: re-authenticating to https://192.168.0.1/: unifi controller: making request: Get "https://192.168.0.1/": dial tcp 192.168.0.1:443: connect: network is unreachable 
```

```yaml
[unpoller] 2026/03/07 14:46:07 [ERROR] metric fetch failed: https://192.168.0.1/: re-authenticating to https://192.168.0.1/: unifi controller: making request: Get "https://192.168.0.1/": dial tcp 192.168.0.1:443: connect: network is unreachable: error initializing inputs 
```

```yaml
[unpoller] 2026/03/07 14:36:22 [ERROR] failed to fetch devices for DHCP lease association: https://192.168.0.1/proxy/network/api/s/default/stat/device: 401 Unauthorized: invalid status code from server 
```

---------------

But today I got 2 new errors which caused my docker to be auto restarted (at 15:53 and 18:17) for the first time. This is the logs for the first time it happened at 15:53 and crashed the app:

```yaml
2026-03-09 15:53:07.061 error [unpoller] 2026/03/09 14:53:07 [ERROR] failed to fetch clients for DHCP lease association: https://192.168.0.1/proxy/network/api/s/default/stat/sta: 401 Unauthorized: invalid status code from server 
2026-03-09 15:53:07.062 error [unpoller] 2026/03/09 14:53:07 [ERROR] failed to fetch devices for DHCP lease association: https://192.168.0.1/proxy/network/api/s/default/stat/device: 401 Unauthorized: invalid status code from server 
2026-03-09 15:53:07.063 error [unpoller] 2026/03/09 14:53:07 [ERROR] failed to fetch networks for DHCP lease association: https://192.168.0.1/proxy/network/api/s/default/rest/networkconf: 401 Unauthorized: invalid status code from server 
2026-03-09 15:53:07.065 [unpoller] panic: runtime error: invalid memory address or nil pointer dereference 
2026-03-09 15:53:07.065 [unpoller] [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8b9c40] 
2026-03-09 15:53:07.065 [unpoller] goroutine 81112 [running]: 
2026-03-09 15:53:07.066 [unpoller] github.com/unpoller/unifi/v5.(*Unifi).GetActiveDHCPLeasesWithAssociations(0x2dfbd14d50e0, {0x2dfbd1334140, 0x1, 0x1}) 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/go/pkg/mod/github.com/unpoller/unifi/v5@v5.18.0/dhcp_leases.go:81 +0x540 
2026-03-09 15:53:07.067 [unpoller] github.com/unpoller/unpoller/pkg/inputunifi.(*InputUnifi).pollController(0x2dfbd1222510, 0x2dfbd1260a00) 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/work/unpoller/unpoller/pkg/inputunifi/collector.go:190 +0x160b 
2026-03-09 15:53:07.067 [unpoller] github.com/unpoller/unpoller/pkg/inputunifi.(*InputUnifi).collectController(0x2dfbd1222510, 0x2dfbd1260a00) 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/work/unpoller/unpoller/pkg/inputunifi/collector.go:79 +0x1be 
2026-03-09 15:53:07.067 [unpoller] github.com/unpoller/unpoller/pkg/inputunifi.(*InputUnifi).Metrics(0x2dfbd1222510, 0x0) 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/work/unpoller/unpoller/pkg/inputunifi/interface.go:254 +0x1aa 
2026-03-09 15:53:07.067 [unpoller] github.com/unpoller/unpoller/pkg/poller.collectMetrics.func1(0x2dfbd19914a0?) 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/work/unpoller/unpoller/pkg/poller/inputs.go:220 +0xcd 
2026-03-09 15:53:07.067 [unpoller] created by github.com/unpoller/unpoller/pkg/poller.collectMetrics in goroutine 81111 
2026-03-09 15:53:07.067 [unpoller] 	/home/runner/work/unpoller/unpoller/pkg/poller/inputs.go:209 +0x6f 
```

--- a/pkg/inputunifi/collector.go
+++ b/pkg/inputunifi/collector.go
@@ -188,13 +188,23 @@ func (u *InputUnifi) pollController(c *Controller) (*poller.Metrics, error) {
 		u.LogDebugf("Found %d SpeedTests entries", len(m.SpeedTests))
 	}
 
-	// Get DHCP leases with associations
-	if m.DHCPLeases, err = c.Unifi.GetActiveDHCPLeasesWithAssociations(sites); err != nil {
-		// Don't fail collection if DHCP leases fail - older controllers may not have this endpoint
-		u.LogDebugf("unifi.GetActiveDHCPLeasesWithAssociations(%s): %v (continuing)", c.URL, err)
-	} else {
-		u.LogDebugf("Found %d DHCPLeases entries", len(m.DHCPLeases))
-	}
+	// Get DHCP leases with associations.
+	// Wrapped in recover so a nil-pointer panic in the library (e.g. when a 401 causes nil devices)
+	// never crashes the poller. See https://github.com/unpoller/unpoller/issues/965
+	func() {
+		defer func() {
+			if r := recover(); r != nil {
+				u.LogErrorf("GetActiveDHCPLeasesWithAssociations panic recovered (see issue #965): %v", r)
+			}
+		}()
+
+		if m.DHCPLeases, err = c.Unifi.GetActiveDHCPLeasesWithAssociations(sites); err != nil {
+			// Don't fail collection if DHCP leases fail - older controllers may not have this endpoint
+			u.LogDebugf("unifi.GetActiveDHCPLeasesWithAssociations(%s): %v (continuing)", c.URL, err)
+		} else {
+			u.LogDebugf("Found %d DHCPLeases entries", len(m.DHCPLeases))
+		}
+	}()
 
 	// Get WAN enriched configuration
 	if m.WANConfigs, err = c.Unifi.GetWANEnrichedConfiguration(sites); err != nil {

